src/capi/azext_capi/custom.py

```diff
@@ -317,9 +320,12 @@ def create_workload_cluster(  # pylint: disable=unused-argument,too-many-argumen
     filename = capi_name + ".yaml"
     end_msg = f'✓ Generated workload cluster configuration at "{filename}"'
     with Spinner(cmd, "Generating workload cluster configuration", end_msg):
-        manifest = template.render(args)
-        with open(filename, "w", encoding="utf-8") as manifest_file:
-            manifest_file.write(manifest)
+        try:
+            manifest = template.render(args)
+            with open(filename, "w", encoding="utf-8") as manifest_file:
+                manifest_file.write(manifest)
+        except UndefinedError as err:
+            raise RequiredArgumentMissingError(f"Could not generate workload cluster configuration. {err}") from err
```

Comment:
Nice, that's the kind of re-raising of Exceptions that Azure CLI wants us to do, so they're handled predictably.